### DISCUSSION TITLE
shortest_path与sssp方法性能差距较大

### DISCUSSION THREAD

**Discussion by inurfuture** [alibaba/GraphScope#1201] [Q&A] [answered]
1.文档中无shortest_path的文档
![image](https://user-images.githubusercontent.com/42594720/146858610-f9c65358-5c4c-480d-81cd-4f1c978667f2.png)
2.相同数据调用sssp和shortest_path，耗时相差很大，如果id使用字符串是否会差别更加大？
![image](https://user-images.githubusercontent.com/42594720/146867614-642f99d1-3e15-482f-9e01-f4f65c28f66c.png)
3.加载文件生成图的方法不支持多文件吗？
![企业微信截图_16400677616458](https://user-images.githubusercontent.com/42594720/146882439-f5d22441-647e-423b-a61e-cdb21e428841.png)

**Comment by acezen** [comment, marked as answer]
hi，@inurfuture
关于第一个问题，目前nx.shortest_path是直接调用的NetworkX的shortest_path, 还没有GraphScope内置版本的实现，所以这个没有更新到builtin algorithm文档中。后续我们会增加一个forward的页面分支，来提供forward app的文档。
关于第二个问题，shortest_path我们现在是直接调用NetworkX的app，它是一个纯python的实现，而sssp是GraphScope提供的内置的c++实现的app，所以在性能上是有差距的，两者还不具备可比性。如果在同一种app上跑，string格式的id和int格式的id理论上性能应该是没区别

**Reply by siyuan0322** [reply]
多文件目前在 OSS/HDFS/S3 是支持的，本地文件还不支持。
